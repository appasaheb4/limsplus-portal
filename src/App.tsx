import React from "react"
import { observer } from "mobx-react"
import * as LibraryComponents from "@lp/library/components"
import { Provider } from "react-redux"
import ReduxToastr from "react-redux-toastr"
import store from "./redux/store/index"
import Routes from "./routes/Routes"
import Contexts from "@lp/library/stores"

const App = observer(() => {
  const rootStore = React.useContext(Contexts.rootStore)
  return (
    <>
      <Provider store={store}>
        <Routes />
        <ReduxToastr
          timeOut={5000}
          newestOnTop={true}
          position="top-right"
          transitionIn="fadeIn"
          transitionOut="fadeOut"
          progressBar
          closeOnToastrClick
        />
      </Provider>
      <LibraryComponents.ToastsContainer
        position={LibraryComponents.ToastsContainerPosition.BOTTOM_RIGHT}
        store={LibraryComponents.ToastsStore}
        className="h-20"
      />
      {rootStore.processLoading && <LibraryComponents.Modal.Loader />}
    </>
  )
})

export default App
