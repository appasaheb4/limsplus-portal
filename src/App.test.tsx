import React from 'react';
import {render} from '@testing-library/react';
import App from './App';

describe('App component', () => {
  it('render App correctly', () => {
    const app = render(<App />);
    expect(app).toMatchSnapshot();
  });
});
