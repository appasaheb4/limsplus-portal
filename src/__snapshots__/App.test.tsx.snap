// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render App correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="vh-50 d-flex container-fluid"
        >
          <div
            class="justify-content-center align-self-center w-100 text-center row"
          >
            <div
              class="spinner-border text-primary"
              role="status"
            >
              <span
                class="visually-hidden"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-live="assertive"
        class="redux-toastr"
      >
        <div>
          <div
            class="top-left"
          />
          <div
            class="top-right"
          />
          <div
            class="top-center"
          />
          <div
            class="bottom-left"
          />
          <div
            class="bottom-right"
          />
          <div
            class="bottom-center"
          />
        </div>
      </div>
      <div
        class="Toastify"
      />
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="vh-50 d-flex container-fluid"
      >
        <div
          class="justify-content-center align-self-center w-100 text-center row"
        >
          <div
            class="spinner-border text-primary"
            role="status"
          >
            <span
              class="visually-hidden"
            >
              Loading...
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-live="assertive"
      class="redux-toastr"
    >
      <div>
        <div
          class="top-left"
        />
        <div
          class="top-right"
        />
        <div
          class="top-center"
        />
        <div
          class="bottom-left"
        />
        <div
          class="bottom-right"
        />
        <div
          class="bottom-center"
        />
      </div>
    </div>
    <div
      class="Toastify"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
